shvvp.c

```diff
@@ -261,8 +261,10 @@ ShvVpLoadCallback (
     if (status != SHV_STATUS_SUCCESS)
     {
         //
-        // Bail out
+        // Bail out, free the allocated per-processor data
+        // Can't free it in Failure, since need to distinguish between the different flows
         //
+        ShvVpFreeData(vpData, 1);
```

Comment:
Actually, please set vpData = NULL on entry, and remove this line and the one that already exists below. Then add if (vpDdata != NULL) { ShvVpFreedata(vpData, 10)) in Failure: label. This is the correct way of handling this for better code quality/readability and maintainability. Thank you!